### PR TITLE
Install cmd

### PR DESCRIPTION
**Pull request by trawler**
**Issue**
Part of #206

**What this PR Includes**
This PR adds the `install` command, that initializes the correct users for k0s server

### CODE REVIEW

File: cmd/install.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+	return nil
+}
+
+func createServerUsers() error {
+	clusterConfig, err := ConfigFromYaml(cfgFile)
+	if err != nil {
+		return err
+	}
+	users := getUserList(*clusterConfig.Install.SystemUsers)
```
Comment (jasmingacic) [inline cmd/install.go:60]: If the config yaml file doesn't contain `installConfig:` the command will fail with:

```
# sudo ./k0s install --config cfg.yaml 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1aef50d]

goroutine 1 [running]:
github.com/k0sproject/k0s/cmd.createServerUsers(0x6b, 0x0)
        /root/src/github.com/k0sproject/k0s/cmd/install.go:60 +0x6d
...
```
Maybe we should check if value of system users are present

File: cmd/user.go [hunk 2]
```diff
@@ -49,10 +49,11 @@ users:
 
 	// userCmd creates new certs and kubeConfig for a user
 	userCmd = &cobra.Command{
-		Use:   "user",
+		Use:   "user [command]",
```
Comment (jasmingacic) [inline cmd/user.go:52]: I know it isn't related to this PR but the command name is a little misleading here. As the command is described as "Manage user access" but in fact it just creates a kubeconfig.

File: internal/util/users.go [hunk 3]
```diff
@@ -37,3 +37,26 @@ func GetGID(name string) (int, error) {
...
+func GetLinuxDist() (string, error) {
+	if runtime.GOOS == "windows" {
+		return "", fmt.Errorf("unsupported OS")
+	}
+	cfg, err := ini.Load("/etc/os-release")
+	if err != nil {
+		fmt.Printf("failed to read file: %v", err)
+	}
+	return cfg.Section("").Key("ID").String(), nil
+}*/
```
Comment (jasmingacic) [inline internal/util/users.go:62]: maybe we could remove this

File: pkg/install/users.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/k0sproject/k0s/internal/util"
+)
+
+// EnsureUser checks if a user exists, and creates it, if it doesn't
+// TODO: we should also consider modifying the user, if the user exists, but with wrong settings
+func EnsureUser(name string, homeDir string) error {
```
Comment (jasmingacic) [inline pkg/install/users.go:17]: The function does what it is supposed to do:
```
# cat /etc/passwd | grep k0s
kube-controller-manager:x:997:997::/var/lib/k0s:/usr/sbin/nologin
etcd:x:996:996::/var/lib/k0s:/usr/sbin/nologin
kube-apiserver:x:995:995::/var/lib/k0s:/usr/sbin/nologin
konnectivity-server:x:994:994::/var/lib/k0s:/usr/sbin/nologin
kube-scheduler:x:993:993::/var/lib/k0s:/usr/sbin/nologin
```

File: cmd/install.go (hunk 1 above)
Comment (trawler) [reply cmd/install.go:60]: Resolved  with https://github.com/k0sproject/k0s/blob/a93e731e3652f53123cd224c01064dd78b7f811c/cmd/helpers.go#L47-L49

File: internal/util/users.go (hunk 3 above)
Comment (trawler) [reply internal/util/users.go:62]: This is a stub that I thought would be useful for future iterations, and I didn't want to remove it :)

File: cmd/install.go (hunk 1 above)
Comment (jasmingacic) [reply cmd/install.go:60]: yes it is resolved

**Review by jasmingacic — APPROVED** [review]

File: cmd/user.go (hunk 2 above)
Comment (trawler) [reply cmd/user.go:52]: I believe this stems from the original issue: #389 and perhaps should be better clarified in a future PR.

**Pull request merged**